src/RedirectMiddleware.php

```diff
@@ -190,7 +191,41 @@ public function modifyRequest(
             $modify['body'] = '';
         }
 
-        $modify['uri'] = $this->redirectUri($request, $response, $protocols);
+        $uri = $this->redirectUri($request, $response, $protocols);
+
+        if ($uri->getHost() && isset($options['idn_conversion']) && ($options['idn_conversion'] !== false)) {
+            $idnOptions = ($options['idn_conversion'] === true) ? IDNA_DEFAULT : $options['idn_conversion'];
+
+            $asciiHost = idn_to_ascii($uri->getHost(), $idnOptions, INTL_IDNA_VARIANT_UTS46, $info);
+            if ($asciiHost === false) {
```

Comment:
What do you think about extracting this logic to a function/helper class? Looks like we have at least 2 places now where it's used, worth extracting.

Reply:
ofc

> if the corrent PR is a solution for issue #2424 then we should consider centralizing this logic..